spec/contact_spec.rb

```diff
@@ -0,0 +1,188 @@
+        let(:status) { 201 }
+
+        it "returns the contact record" do
+          result = subject.create_or_update
+
+          expect(result).to be_a Maropost::Contact
+          expect(result.email).to eq email
+          expect(result.travel_especials).to eq opt_in
+          expect(result.vr_reminder_autocall).to eq opt_in
+          expect(result.business_vehicle_reminder).to eq business_name
+          expect(result.errors).to_not be_present
+        end
+      end
+
+      context "error" do
+        let(:body) { "{}" }
+        let(:status) { 422 }
+
+        it "returns a contact record with errors" do
+          result = subject.create_or_update
+          expect(result).to be_a Maropost::Contact
+          expect(result.email).to eq email
+          expect(result.errors[:base].first).to eq "Unable to create or update contact"
+        end
+      end
+    end
+
+    context "updating an existing record" do
+      let(:get_body) { File.read File.join("spec", "fixtures", "contacts", "get_response.json") }
+      let(:get_status) { 200 }
+      let(:id) { 741000000 }
+
+      before(:each) do
+        stub_request(:put, "#{Maropost.configuration.api_url}/contacts/#{id}.json").to_return(
+          status: status,
+          body: body
+        )
+      end
+
+      context "success" do
+        let(:body) { File.read File.join("spec", "fixtures", "contacts", "create_response.json") }
+        let(:status) { 201 }
+
+        it "returns the contact record" do
+          result = subject.create_or_update
+
+          expect(result).to be_a Maropost::Contact
+          expect(result.email).to eq email
+          expect(result.travel_especials).to eq opt_in
+          expect(result.vr_reminder_autocall).to eq opt_in
+          expect(result.business_vehicle_reminder).to eq business_name
+          expect(result.errors).to_not be_present
+        end
+      end
+
+      context "error" do
+        let(:body) { "{}" }
+        let(:status) { 422 }
+
+        it "returns a contact record with errors" do
+          result = subject.create_or_update
+          expect(result).to be_a Maropost::Contact
+          expect(result.email).to eq email
+          expect(result.errors[:base].first).to eq "Unable to create or update contact"
+        end
+      end
+    end
+  end
+
+  describe "#update_email" do
+    let(:subject) { Maropost::Contact.new(data) }
+    let(:id) { 741000000 }
+    let(:new_email) { "wafflezone@example.com" }
+    let(:data) {
+      {
+        email: new_email,
+        id: id
+      }
+    }
+
+    before(:each) do
+      stub_request(:put, "#{Maropost.configuration.api_url}/contacts/#{id}.json").to_return(
+        status: status,
+        body: body
+      )
+    end
+
+    context "success" do
+      let(:body) { File.read File.join("spec", "fixtures", "contacts", "update_email_response.json") }
+      let(:status) { 201 }
+
+      it "returns the contact record" do
+        result = subject.update_email(new_email)
+
```

Comment:
empty line unless you are leaving it to separate is from the expect lines (same for the above comments of empty lines)